.gitattributes

```
@@ -1 +1,2 @@
 * text eol=lf
```

```suggestion
* text=auto
```
This is probably better.

---

afaik this was intentionally set to lf so that all files are lf, otherwise it may differ?

---

> text
> 
> This attribute marks the path as a text file, which enables end-of-line conversion: When a matching file is added to the index, the file’s line endings are normalized to LF in the index. Conversely, when the file is copied from the index to the working directory, its line endings may be converted from LF to CRLF depending on the eol attribute, the Git config, and the platform (see explanation of eol below).
> 

Text does normalization to LF automatically.

> Set to string value "auto"
> 
> When text is set to "auto", Git decides by itself whether the file is text or binary. If it is text and the file was not already in Git with CRLF endings, line endings are converted on checkin and checkout as described above. Otherwise, no conversion is done on checkin or checkout.
> 

Auto automatically decides if text or binary, and performs normalization to LF automatically on text files, unless it's already in git with CRLF ending.

So basically, this line is everything you need, if some binaries are not recognized, or file has no extension, you can manually specify that it is binary with this pattern `*.jpg binary`

here is the example autoconfig:
```
# This will do normalization to LF on index (staging area) 
* text=auto

*.java   text
*.html   text
*.css    text

# Explicit for linux files
*.sh     text eol=lf

# Explicit stuff for win files
*.bat    text eol=crlf

# Just to make sure git doesn't corrupt our binaries accidentally, it's a good practice.
*.gif binary
*.jpg binary
*.jpeg binary
*.png binary

# No extensions, so we set them manually
/gradlew           text eol=lf
```

---

```
*.java   text
*.html   text
*.css    text
```

do we explicitly need to add this